### ISSUE TITLE
Seed is not fixed in LightGBM

### ISSUE THREAD

**Issue by razmikmelikbekyan** [awslabs/autogluon#292]
https://github.com/awslabs/autogluon/blob/9a214c5e97363447c321e4e307d8faf0c861dd65/autogluon/utils/tabular/ml/models/lgb/lgb_model.py#L78

**Comment by Innixma** [comment]
This indeed should mean that results will differ each run, however I can't seem to make them differ even when trying.

Setting seed value to different values doesn't seem to change the results. Not sure why.

I even tried setting the other seed parameters:

```
if seed_val is not None:
     train_params['params']['seed'] = seed_val
     train_params['params']['feature_fraction_seed'] = seed_val
     train_params['params']['bagging_seed'] = seed_val
```

According to their documentation, this should result in different behavior: [https://lightgbm.readthedocs.io/en/latest/Parameters.html#seed](https://lightgbm.readthedocs.io/en/latest/Parameters.html#seed)

But it is the same result every time, early stopping on the same iteration. (Tested on my Mac)

Not sure if this is somehow an issue on AutoGluon's side of an issue from LightGBM.

**Comment by razmikmelikbekyan** [comment]
It seems that `seed` parameter is used for generation other seeds, e.g. `data_random_seed`, `feature_fraction_seed`, etc. And all of these other seeds have `not None` default values, that is why the results are deterministic. So either we can close this issue without changing anything, either for safety we can put some default value to `seed`, which will not change the other seeds anyway (unless there will be change in LightGBM side), because this seed is unused in favor of default values of other seeds: https://lightgbm.readthedocs.io/en/latest/Parameters.html#seed .

**Comment by Innixma** [comment]
Yes, that would be the expected behavior (that the default values for other seeds are set, thus ignoring generic seed). However, it seems even changing the other seed values doesn't have an effect. Very odd.

I don't think it's particularly a problem, it would only be an issue if a user specified these parameters and was expecting them to affect things. But at the end of it all, I think it is a LightGBM problem, and I don't see a reason why a user would ever care about the seeds of LightGBM in AutoGluon, there are much more impactful parameters to be specifying.

**Comment by Innixma** [comment]
Created #297 to update to 0

**Comment by Innixma** [comment]
Merged #297
